core/src/qos_service.rs

```diff
@@ -0,0 +1,239 @@
+//! Producer quality of service
```

Comment:
Producer isn't a term we defined. Should this say Leader? Or maybe just flesh this out to be more descriptive?

Reply:
added few words of description